Makefile

```diff
@@ -46,7 +46,7 @@ release: tools
 
 static: tools docs
 	cargo fmt -- --check
-	cargo clippy $(FEATURES)
+	cargo clippy $(FEATURES) -- -Dwarnings
```

Comment:
Can you add this to the `build --all-targets` in the test target too? I'm not sure clippy checks everything like benchmarks, etc, right?

Reply:
It doesn't recognize the option, and it seems like this is still an open request for Cargo#8424. Yes this might miss some things but I'm not sure what the right solution would be. Maybe we would have to put `#![deny(warnings)]` in test code?

Reply:
This wouldn't be making anything worth though right? Because test/benchmark targets would be built separately and not respect the directive in the lib/bin anyway? So we're making CI more strict and it will definitely check the code we removed the directive from. I'm tempted to leave figuring out erroring on lints for benchmarks as a problem to solve